### ISSUE TITLE
IntelLLVM minimum version check uses the version of underlying Clang

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos#4758]
During configure with dpcpp (IntelLLVM), Kokkos develop now requires version 2022.0.0 or higher (no issue with that). But when CMake computes ``CMAKE_CXX_COMPILER_VERSION``, it appears to get the underlying Clang version (14.0.0) and then Kokkos says "14 < 2022, so this is too old".

Error output:
```
CMake Error at cmake/kokkos_compiler_id.cmake:164 (MESSAGE):
  Compiler not supported by Kokkos.  Required compiler versions:

      Clang        4.0.0 or higher
      GCC          5.3.0 or higher
      Intel       17.0.0 or higher
      IntelLLVM 2022.0.0 or higher
      NVCC        9.2.88 or higher
      HIPCC        4.5.0 or higher
      PGI           17.4 or higher

  Compiler: IntelLLVM 14.0.0

Call Stack (most recent call first):
  cmake/kokkos_tribits.cmake:212 (INCLUDE)
  CMakeLists.txt:198 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
```

Configured with
```
cmake \
-DCMAKE_CXX_COMPILER=dpcpp \
-DKokkos_ENABLE_SYCL=ON \
-DKokkos_ARCH_INTEL_XEHP=ON \
-DCMAKE_CXX_EXTENSIONS=OFF \
-DCMAKE_CXX_STANDARD=17 \
../kokkos
```

The version coming back as 14.0.0 isn't a Kokkos problem, this hello world CMakeLists.txt:
```
project(asdf)
MESSAGE(STATUS "Hello, compiler version is ${CMAKE_CXX_COMPILER_VERSION}")
```
when configured with ``cmake -DCMAKE_CXX_COMPILER=dpcpp ..`` will also print 14.0.0.

**Comment by ajpowelsnl** [comment]
Thanks for this @brian-kelley -- Just to be clear, here, we want to make `CMAKE_CXX_COMPILER_VERSION` return the llvm compiler version, vs. the underlying clang you mention?

**Comment by brian-kelley** [comment]
@ajpowelsnl I don't think it's possible to change the internal CMake variable ``CMAKE_CXX_COMPILER_VERSION`` directly, but you could override ``KOKKOS_CXX_COMPILER_VERSION``. It looks like there are already paths in kokkos_compiler_id.cmake that do that for Cuda and HIP.

**Comment by masterleinad** [comment]
Hmm... I can't really reproduce. For me, the check works with both `icpx` and `dpcpp` correctly (using `CMake 3.22.1`). What is your `CMake` version?

**Comment by brian-kelley** [comment]
@masterleinad Ah you're right, I was using 3.19 but 3.20 and up doesn't have this problem. Thanks!

**Comment by ajpowelsnl** [comment]
Morning @masterleinad -- same question here: will this work go into Kokkos 3.6?

**Comment by masterleinad** [comment]
Yes, it should IMHO. I marked the corresponding pull request with "Blocks Promotion".

**Comment by ajpowelsnl** [comment]
Hi @masterleinad and @brian-kelley -- Are we good to go with the fix on this bug?  If so, may I close this issue?

**Comment by masterleinad** [comment]
@ajpowelsnl We will close all issues that have the `InDevelop` label once the release is finalized.

**Comment by PhilMiller** [comment]
Fix released in 3.6
